### PR TITLE
added feature for adding reference links

### PR DESCRIPTION
**Pull request by rohansharmasitoula**

### CODE REVIEW

File: db/upgrade.php [hunk 1]
```diff
@@ -39,19 +39,34 @@ function xmldb_quizaccess_proctor_upgrade($oldversion) {
...
-
-        // Changing the default of field tsbenabled on table quizaccess_proctor to 0.
+//    if ($oldversion < 2023080809) {
+//
+//        // Changing the default of field tsbenabled on table quizaccess_proctor to 0.
+//        $table = new xmldb_table('quizaccess_proctor');
+//        $field = new xmldb_field('tsbenabled', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, '0', 'proctortype');
+//
+//        // Launch change of default for field tsbenabled.
+//        $dbman->change_field_default($table, $field);
```
Comment (devang1281) [inline db/upgrade.php]: @rohansharmasitoula Why is this changed

File: db/upgrade.php [hunk 2]
```diff
@@ -39,19 +39,34 @@ function xmldb_quizaccess_proctor_upgrade($oldversion) {
...
+//
+//        // Proctor savepoint reached.
+//        upgrade_plugin_savepoint(true, 2023080809, 'quizaccess', 'proctor');
+//    }
+    if ($oldversion < 2023080909) {
```
Comment (devang1281) [inline db/upgrade.php]: @rohansharmasitoula Keep the version same as version.php and install.xml

File: db/upgrade.php [hunk 3]
```diff
@@ -39,19 +39,34 @@ function xmldb_quizaccess_proctor_upgrade($oldversion) {
...
-        // Launch change of default for field tsbenabled.
-        $dbman->change_field_default($table, $field);
+        // Conditionally launch add field reference_link.
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
 
         // Proctor savepoint reached.
-        upgrade_plugin_savepoint(true, 2023080101, 'quizaccess', 'proctor');
+        upgrade_plugin_savepoint(true, 2023080909, 'quizaccess', 'proctor');
```
Comment (devang1281) [inline db/upgrade.php]: @rohansharmasitoula Keep the version same as version.php and install.xml

File: version.php [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version = 2023080701;
+$plugin->version = 2023080807;
 $plugin->requires = 2020061500;
 $plugin->release = '1.0.0 (Build: 2023080701)';
```
Comment (devang1281) [inline version.php]: Fix the Version here

**Review by devang1281 — CHANGES_REQUESTED** [review]

File: db/upgrade.php (hunk 2 above)
Comment (rohansharmasitoula) [reply db/upgrade.php]: done

File: db/upgrade.php (hunk 1 above)
Comment (rohansharmasitoula) [reply db/upgrade.php]: done

File: db/upgrade.php (hunk 3 above)
Comment (rohansharmasitoula) [reply db/upgrade.php]: done

File: version.php (hunk 4 above)
Comment (rohansharmasitoula) [reply version.php]: fixed

**Review by devang1281 — APPROVED** [review]
LGTM
